cloudinit/net/networkd.py

```diff
@@ -242,6 +242,20 @@ def _render_content(self, ns):
                 name = iface['name']
                 # network state doesn't give dhcp domain info
                 # using ns.config as a workaround here
+
+                # Check to see if this interface matches against an interface
+                # from the network state that specified a set-name directive.
+                # If there is a device with a set-name directive and it has
+                # set-name value that matches the current name, then update the
+                # current name to the device's name. That will be the valeu in
+                # the ns.config['ethernets'] dict below.
+                for dev_name, dev_cfg in ns.config['ethernets'].items():
```

Comment:
Nit: Lines 253 through 258 seems a bit hard to read to me. Redundant interleaved assignment/if statements makes the intention of this less clear. I think something like the following is easier to read (for me at least) and is logically equivalent in half the number of lines.
```
                    if dev_cfg.get('set-name') == name:
                        name = dev_name
                        break
```

Maybe others feel different about style.